backend/src/main/java/com/soen341/instagram/service/impl/RegistrationService.java

```diff
@@ -0,0 +1,62 @@
+package com.soen341.instagram.service.impl;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.exception.account.AccountExistException;
+import com.soen341.instagram.exception.account.InvalidEmailException;
+import com.soen341.instagram.model.Account;
+
+@Service("registrationService")
+public class RegistrationService
+{
+	@Autowired
+	private AccountRepository accountRepository;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	public void createNewAccount(final Account account) throws ParseException
+	{
+		if (accountRepository.findByUsername(account.getUsername()) == null
+				&& accountRepository.findByEmail(account.getEmail()) == null)
```

Comment:
Just to make sure: we're only letting one email be associated with one account, right? No user can have multiple accounts for the same email?

Reply:
Also, the email check should be done separately, because in this case, if someone tries to register with an already existing email (but with a valid and non-used username), it will still go to the else branch and throw an `AccountExistException`.

Reply:
@emilie-martin Yes, a user shpuld only have one email associated to its account.